src/magic.cpp

```diff
@@ -989,6 +1036,21 @@ bool spell::cast_all_effects( const Creature &source, const tripoint &target ) c
     return success;
 }
 
+void spell::set_obj_name( std::string new_name )
+{
+    obj_name = new_name;
+}
+
+const char *spell::get_obj_name() const
+{
+    return obj_name.c_str();
```

Comment:
Please don't use `const char*` at all, that can easily cause so much trouble. Expose the name as `const std::string &` instead.

Reply:
will do. This function will probably be removed.